syntax = "proto3";

package MyMQ;

import "msg.proto";

message OpenChannelRequest
{
    string rid = 1;
    string cid = 2;
};

message CloseChannelRequest
{
    string rid = 1;
    string cid = 2;
};

message DeclareExchangeRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    ExchangeType exchange_type = 4;
    bool durable = 5;
    bool auto_delete = 6;
    map<string, string> args = 7;
};

message DeleteExchangeRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
};

message DeleteQueueRequest
{
    string rid = 1;
    string cid = 2;
    string queue_name = 3;
};

// 定义队列
message DeclareQueueRequest
{
    string rid = 1;
    string cid = 2;
    string queue_name = 3;
    bool exclusive = 4;
    bool durable = 5;
    bool auto_delete = 6;
    map<string, string> args = 7;
};

message QueueBindRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string queue_name = 4;
    string binding_key = 5;
}

// 队列
message QueueUnBindRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string queue_name = 4;
};

// 订阅发布
message BasicPublishRequest
{
    string rid = 1;
    string cid = 2;
    string exchange_name = 3;
    string body = 4;
    BasicProperties properties = 5;
};

// 消息确认
message BasicAckRequest
{
    string cid = 1;
    string rid = 2;
    string queue_name = 3;
    string message_id = 4;
};

// 队列的订阅
message BasicConsumeRequest
{
    string rid = 1;
    string cid = 2;
    string consumer_tag = 3;
    string queue_name = 4;
    bool auto_ack = 5;
};

// 信息推送
message BasicConsumeResponse
{
    string cid = 1;
    string consumer_tag = 2;
    string body = 3;
    BasicProperties properties = 4;
};

// 订阅的取消
message BasicCancelRequest 
{
    string rid = 1;
    string cid = 2;
    string consumer_tag = 3;
    string queue_name = 4;
};


// 通用响应
message BasicCommonResponse
{
    string rid = 1;
    string cid = 2;
    bool ok = 3;
};